Added redhat-subscriptions, manage_repositories and update-and-instal…

kbidarkar opened the pull request:
…l role

Signed-off-by: Kedar Bidarkar <kbidarka@redhat.com>

kbidarkar committed:
Added redhat-subscriptions, manage_repositories, update-and-install role

*) redhat-subscriptions role deals with registration and
   subscribing content
*) manage-repositories role deals with enabling and disabling
   of repositories.
*) update-os-packages role deals with updating all the installed
   packages.
*) install-satellite6 role deals with installing the sat6 packages.

The idea os separating out update-os-packages and install-satellite6
was to make use of update-os-packages elsewhere too, like in other
playbooks like upgrade.

Signed-off-by: Kedar Bidarkar <kbidarka@redhat.com>

ehelms commented on roles/manage-repositories/tasks/enable_repositories.yml:
```diff
@@ -0,0 +1,31 @@
+  when:
+    - satellite_distribution == "internal_repo"
+  yum_repository:
+    name: "{{ satellite_reponame }}"
```
Best practices say to name your variables as scoped by the role itself to prevent clashes. So this would translate to `manage_repositories_satellite_reponame`

ehelms commented on roles/redhat-subscriptions/defaults/main.yml:
```diff
@@ -0,0 +1,4 @@
+---
+# defaults file for redhat-subscriptions
+rhsm_username: user@redhat.com
+rhsm_password: QMNBjdkfIJHDkka
```
I assume this is fake? Maybe make it something that reads slightly better that way? Like `fake-password`

kbidarkar replied:
yup, will do that

kbidarkar replied:
Yes, it fake. Just want to mention this explicitly.

ehelms commented on roles/redhat-subscriptions/tasks/activation_key.yml:
```diff
@@ -0,0 +1,14 @@
+- name: Register with activationkey and consume subscriptions from Dogfood Server.
+  redhat_subscription:
+    state: present
+    activationkey: "{{ dogfood_ak }}"
```
Minor suggestion to rename this to `dogfood_akey`. I would also argue for not tieing this to dogfood and follow best practices. That is, change this to be `redhat_subscriptions_activation_key` and `redhat_subscriptions_organization`. That allows the role to be more generic.

ehelms commented on roles/redhat-subscriptions/tests/test.yml:
```diff
@@ -0,0 +1,4 @@
+---
+- hosts: sat63
+  roles:
+    - partition-disk
```
Should the tests be using the redhat-subscriptions role?

kbidarkar replied:
I missed this, a typo, will update it. I was more looking into the directory structure and overlooked this.

ehelms commented on roles/update-and-install/tasks/main.yml:
```diff
@@ -0,0 +1,14 @@
+---
```
We split system update from install of the software and found this to be useful in a few scenarios to de-couple those two actions. Our update system role:

https://github.com/theforeman/forklift/tree/master/roles/update_os_packages/tasks

kbidarkar replied:
ACK will update this, thanks for sharing the experience.

ehelms commented on vars/satellite_common.yml:
```diff
@@ -0,0 +1,27 @@
+rhsm_pool_id2: 8a85f98960dbf6510160df23e3367451
+
+# Dogfood URL
+dogfood_url: "http://dogfood.example.com"
```
Per earlier comments, I'd change this just to be not specific to dogfood.

kbidarkar replied:
Ok, will make it `repository_url`.

kbidarkar replied:
or as per the best-practices.

ehelms commented on vars/satellite_common.yml:
```diff
@@ -0,0 +1,27 @@
+# Satellite Employee subscriptions for all other Product/Repos.
+rhsm_pool_id1: 8a85f981519abf020151a22d3c387f2a
+# Employee SKU for 'Red Hat Ansible Engine Product'.
+rhsm_pool_id2: 8a85f98960dbf6510160df23e3367451
```
I assume it's fine to put pool IDs out on git not wrapped in a secret?

kbidarkar replied:
I assume it is ok, those anyways won't be available unless one uses proper creds.

ogajduse replied:
I agree with @ehelms.
Do we plan to have some internal repository with some custom or internal configuration in it? If yes, I think that putting these IDs in should be a good start.

kbidarkar replied:
@ogajduse those values are not sensitive. 

But answering about internal stuff, we will have another repo internally for sensitive stuff.

Any sensitive stuff would be a one-time configuration at one's own end. 

satellite_common_local.yml file, which we would not add to version control.

I will demo stuff next week regarding the workflow, which would help.

kbidarkar replied:
`cp  satellite_common.yml satellite_common_local.yml`

@ogajduse , good you asked, will do some changes which would make it very clear. Thanks.

kbidarkar replied:
Why is pool-id not sensitive?
Explanation: you can pass in either `pool` or `pool-id` here, for reasons of speed I have used `pool-id`
pool == "Subscription Name" for example: `Red Hat Satellite Employee Subscription` or `Employee SKU`
pool-id == It's corresponding ID.
So, clearly I feel it is not sensitive information in this case.

kbidarkar replied:
https://docs.ansible.com/ansible/2.5/modules/redhat_subscription_module.html
From the above link, 
`pool_ids: Specify subscription pool IDs to consume. Prefer over pool when possible as it is much faster.`

ehelms commented on roles/redhat-subscriptions/README.md:
```diff
@@ -0,0 +1,45 @@
+
+Requires rhsm_username and rhsm_password to be set in vars/satellite_common.yml if using internal_repo or CDN.
+
+Requires dogfood_ak and dogfood_org to be set in vars/satellite_63.yml if using internal_ak.
```
I would not include the specific mention that it requires `vars/satellite_63.yml` and the above `satellite_common` because these are roles and they don't care what vars file you use.

kbidarkar commented:
Below is the output after execution:

```
 kbid  (e) ansible3  ~  git_world  ansible-satellite6  ansible-playbook playbooks/install/satellite_63.yml

PLAY [sat63] ******************************************************************************************************************************************************************************************************

TASK [Gathering Facts] ********************************************************************************************************************************************************************************************
ok: [sat63-rhel7]

TASK [partition-disk : include_tasks] *****************************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/partition-disk/tasks/remove_home_partition.yml for sat63-rhel7

TASK [partition-disk : Unmount /home partition] *******************************************************************************************************************************************************************
ok: [sat63-rhel7]

TASK [partition-disk : Remove /home entry from /etc/fstab] ********************************************************************************************************************************************************
ok: [sat63-rhel7]

TASK [partition-disk : Remove /home Logical Volume] ***************************************************************************************************************************************************************

TASK [partition-disk : include_tasks] *****************************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/partition-disk/tasks/extend_root_partition.yml for sat63-rhel7

TASK [partition-disk : Extend root partition LV to all available VG space] ****************************************************************************************************************************************
ok: [sat63-rhel7] => (item={'key': 'root', 'value': {'size_g': '1830.38', 'vg': 'rhel_intel-wildcatpass-01'}})

TASK [partition-disk : Resize the root filesystem] ****************************************************************************************************************************************************************
ok: [sat63-rhel7] => (item={'key': 'root', 'value': {'size_g': '1830.38', 'vg': 'rhel_intel-wildcatpass-01'}})

TASK [redhat-subscriptions : include_tasks] ***********************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/redhat-subscriptions/tasks/username_and_password.yml for sat63-rhel7

TASK [redhat-subscriptions : Register and Subscribe RHEL Instance to RHN.] ****************************************************************************************************************************************
changed: [sat63-rhel7]

TASK [redhat-subscriptions : include_tasks] ***********************************************************************************************************************************************************************

TASK [manage-repositories : include_tasks] ************************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/manage-repositories/tasks/disable_repositories.yml for sat63-rhel7

TASK [manage-repositories : Install Yum Utils] ********************************************************************************************************************************************************************
ok: [sat63-rhel7]

TASK [manage-repositories : Disable all RHSM repositories] ********************************************************************************************************************************************************
changed: [sat63-rhel7]

TASK [manage-repositories : Disable all Beaker Repos] *************************************************************************************************************************************************************
changed: [sat63-rhel7]

TASK [manage-repositories : include_tasks] ************************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/manage-repositories/tasks/enable_repositories.yml for sat63-rhel7

TASK [manage-repositories : Enable all the required repositories for Satellite6 Installation.] ********************************************************************************************************************
changed: [sat63-rhel7] => (item=rhel-7-server-rpms)
changed: [sat63-rhel7] => (item=rhel-7-server-optional-rpms)
changed: [sat63-rhel7] => (item=rhel-7-server-extras-rpms)
changed: [sat63-rhel7] => (item=rhel-7-server-satellite-maintenance-6-rpms)
changed: [sat63-rhel7] => (item=rhel-7-server-ansible-2-rpms)
changed: [sat63-rhel7] => (item=rhel-server-rhscl-7-rpms)

TASK [manage-repositories : Enable Satellite6 repository.] ********************************************************************************************************************************************************
changed: [sat63-rhel7]

TASK [manage-repositories : Add Satellite 6 repository] ***********************************************************************************************************************************************************

TASK [update-and-install : Update all packages before installing Satellite6] **************************************************************************************************************************************
ok: [sat63-rhel7]

TASK [update-and-install : Install the latest version of Satellite6] **********************************************************************************************************************************************
changed: [sat63-rhel7]

PLAY RECAP ********************************************************************************************************************************************************************************************************
sat63-rhel7                : ok=18   changed=6    unreachable=0    failed=0   

```

kbidarkar commented:
Below is the output after installation via `satellite_distribution:cdn`. All packages installed successfully.
```
[root@sat63-rhel7 ~]# rpm -qav | grep -i satellite
satellite-common-6.3.2-1.el7sat.noarch
satellite-cli-6.3.2-1.el7sat.noarch
satellite-installer-6.3.0.12-1.el7sat.noarch
satellite-6.3.2-1.el7sat.noarch
tfm-rubygem-foreman_theme_satellite-1.0.4.19-1.el7sat.noarch
```

kbidarkar commented on playbooks/install/satellite_63.yml:
```diff
@@ -1,4 +1,10 @@
 ---
 - hosts: sat63
+  vars_files:
+    - ../../vars/satellite_common.yml
```
This should be `../../vars/satellite_common.local.yml` which would not be added to version control. Will fix these.

Need to add the workflow to the README file about, `cp satellite_common.yml satellite_common.local.yml`

kbidarkar replied:
One good thing would be to add `satellite_common.local.yml` to `.gitignore`  to avoid any accidental addition of it.

kbidarkar replied:
populating `satellite_common.local.yml` would just be a one-time effort.

kbidarkar commented:
Install Satellite 6.3 from CDN:
```
 kbid  (e) ansible3  ~  git_world  ansible-satellite6  ansible-playbook playbooks/install/satellite_63.yml 

PLAY [sat63] *****************************************************************************************************************************************************************************************************************************************************************************

TASK [Gathering Facts] *******************************************************************************************************************************************************************************************************************************************************************
ok: [sat63-rhel7]

TASK [partition-disk : include_tasks] ****************************************************************************************************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/partition-disk/tasks/remove_home_partition.yml for sat63-rhel7

TASK [partition-disk : Unmount /home partition] ******************************************************************************************************************************************************************************************************************************************
ok: [sat63-rhel7]

TASK [partition-disk : Remove /home entry from /etc/fstab] *******************************************************************************************************************************************************************************************************************************
ok: [sat63-rhel7]

TASK [partition-disk : Remove /home Logical Volume] **************************************************************************************************************************************************************************************************************************************

TASK [partition-disk : include_tasks] ****************************************************************************************************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/partition-disk/tasks/extend_root_partition.yml for sat63-rhel7

TASK [partition-disk : Extend root partition LV to all available VG space] ***************************************************************************************************************************************************************************************************************
ok: [sat63-rhel7] => (item={'key': 'root', 'value': {'size_g': '1830.38', 'vg': 'rhel_intel-wildcatpass-01'}})

TASK [partition-disk : Resize the root filesystem] ***************************************************************************************************************************************************************************************************************************************
ok: [sat63-rhel7] => (item={'key': 'root', 'value': {'size_g': '1830.38', 'vg': 'rhel_intel-wildcatpass-01'}})

TASK [redhat-subscriptions : include_tasks] **********************************************************************************************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/redhat-subscriptions/tasks/username_and_password.yml for sat63-rhel7

TASK [redhat-subscriptions : Register and Subscribe RHEL Instance to RHN.] ***************************************************************************************************************************************************************************************************************
changed: [sat63-rhel7]

TASK [redhat-subscriptions : include_tasks] **********************************************************************************************************************************************************************************************************************************************

TASK [manage-repositories : include_tasks] ***********************************************************************************************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/manage-repositories/tasks/disable_repositories.yml for sat63-rhel7

TASK [manage-repositories : Install Yum Utils] *******************************************************************************************************************************************************************************************************************************************
ok: [sat63-rhel7]

TASK [manage-repositories : Disable all RHSM repositories] *******************************************************************************************************************************************************************************************************************************
changed: [sat63-rhel7]

TASK [manage-repositories : Disable all Beaker Repos] ************************************************************************************************************************************************************************************************************************************
changed: [sat63-rhel7]

TASK [manage-repositories : include_tasks] ***********************************************************************************************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/manage-repositories/tasks/enable_repositories.yml for sat63-rhel7

TASK [manage-repositories : Enable all the required repositories for Satellite6 Installation.] *******************************************************************************************************************************************************************************************
changed: [sat63-rhel7] => (item=rhel-7-server-rpms)
changed: [sat63-rhel7] => (item=rhel-7-server-optional-rpms)
changed: [sat63-rhel7] => (item=rhel-7-server-extras-rpms)
changed: [sat63-rhel7] => (item=rhel-7-server-satellite-maintenance-6-rpms)
changed: [sat63-rhel7] => (item=rhel-7-server-ansible-2-rpms)
changed: [sat63-rhel7] => (item=rhel-server-rhscl-7-rpms)

TASK [manage-repositories : Enable Satellite6 repository.] *******************************************************************************************************************************************************************************************************************************
changed: [sat63-rhel7]

TASK [manage-repositories : Add Satellite 6 repository] **********************************************************************************************************************************************************************************************************************************

TASK [update-os-packages : Update all installed packages before installing Satellite6] ***************************************************************************************************************************************************************************************************
ok: [sat63-rhel7]

TASK [install-satellite6 : Install the latest version of Satellite6] *********************************************************************************************************************************************************************************************************************
ok: [sat63-rhel7]

PLAY RECAP *******************************************************************************************************************************************************************************************************************************************************************************
sat63-rhel7                : ok=18   changed=5    unreachable=0    failed=0   

```

kbidarkar commented:
Install `Sat6.4` from `internal_ak`.
```
 kbidarka  (e) ansible3  ~  git_world  ansible-satellite6  ansible-playbook playbooks/install/satellite_64.yml 

PLAY [sat64] ******************************************************************************************************************************************************************************************************

TASK [Gathering Facts] ********************************************************************************************************************************************************************************************
ok: [sat64-rhel7]

TASK [partition-disk : include_tasks] *****************************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/partition-disk/tasks/remove_home_partition.yml for sat64-rhel7

TASK [partition-disk : Unmount /home partition] *******************************************************************************************************************************************************************
ok: [sat64-rhel7]

TASK [partition-disk : Remove /home entry from /etc/fstab] ********************************************************************************************************************************************************
ok: [sat64-rhel7]

TASK [partition-disk : Remove /home Logical Volume] ***************************************************************************************************************************************************************

TASK [partition-disk : include_tasks] *****************************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/partition-disk/tasks/extend_root_partition.yml for sat64-rhel7

TASK [partition-disk : Extend root partition LV to all available VG space] ****************************************************************************************************************************************
ok: [sat64-rhel7] => (item={'key': 'root', 'value': {'size_g': '128.00', 'vg': 'rhel_mgmt5'}})

TASK [partition-disk : Resize the root filesystem] ****************************************************************************************************************************************************************
ok: [sat64-rhel7] => (item={'key': 'root', 'value': {'size_g': '128.00', 'vg': 'rhel_mgmt5'}})

TASK [redhat-subscriptions : include_tasks] ***********************************************************************************************************************************************************************

TASK [redhat-subscriptions : include_tasks] ***********************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/redhat-subscriptions/tasks/activation_key.yml for sat64-rhel7

TASK [redhat-subscriptions : Install katello-ca-consumer-latest] **************************************************************************************************************************************************
ok: [sat64-rhel7]

TASK [redhat-subscriptions : Register with activationkey and consume subscriptions from Dogfood Server.] **********************************************************************************************************
ok: [sat64-rhel7]

TASK [manage-repositories : include_tasks] ************************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/manage-repositories/tasks/disable_repositories.yml for sat64-rhel7

TASK [manage-repositories : Install Yum Utils] ********************************************************************************************************************************************************************
ok: [sat64-rhel7]

TASK [manage-repositories : Disable all RHSM repositories] ********************************************************************************************************************************************************

TASK [manage-repositories : Disable all Beaker Repos] *************************************************************************************************************************************************************
changed: [sat64-rhel7]

TASK [manage-repositories : include_tasks] ************************************************************************************************************************************************************************

TASK [update-os-packages : Update all installed packages before installing Satellite6] ****************************************************************************************************************************
changed: [sat64-rhel7]

TASK [install-satellite6 : Install the latest version of Satellite6] **********************************************************************************************************************************************
changed: [sat64-rhel7]

RUNNING HANDLER [update-os-packages : yum-clean-all] **************************************************************************************************************************************************************
changed: [sat64-rhel7]

PLAY RECAP ********************************************************************************************************************************************************************************************************
sat64-rhel7                : ok=16   changed=4    unreachable=0    failed=0  
```

kbidarkar commented:
Install Satellite 6.3 from CDN:
```
 kbid  (e) ansible3  ~  git_world  ansible-satellite6  ansible-playbook playbooks/install/satellite_63.yml

PLAY [sat63] ******************************************************************************************************************************************************************************************************

TASK [Gathering Facts] ********************************************************************************************************************************************************************************************
ok: [sat63-rhel7]

TASK [partition-disk : include_tasks] *****************************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/partition-disk/tasks/remove_home_partition.yml for sat63-rhel7

TASK [partition-disk : Unmount /home partition] *******************************************************************************************************************************************************************
changed: [sat63-rhel7]

TASK [partition-disk : Remove /home entry from /etc/fstab] ********************************************************************************************************************************************************
changed: [sat63-rhel7]

TASK [partition-disk : Remove /home Logical Volume] ***************************************************************************************************************************************************************
changed: [sat63-rhel7] => (item={'key': 'home', 'value': {'size_g': '80.85', 'vg': 'rhel_qe-blade-04'}})

TASK [partition-disk : include_tasks] *****************************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/partition-disk/tasks/extend_root_partition.yml for sat63-rhel7

TASK [partition-disk : Extend root partition LV to all available VG space] ****************************************************************************************************************************************
changed: [sat63-rhel7] => (item={'key': 'root', 'value': {'size_g': '50.00', 'vg': 'rhel_qe-blade-04'}})

TASK [partition-disk : Resize the root filesystem] ****************************************************************************************************************************************************************
changed: [sat63-rhel7] => (item={'key': 'root', 'value': {'size_g': '50.00', 'vg': 'rhel_qe-blade-04'}})

TASK [redhat-subscriptions : include_tasks] ***********************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/redhat-subscriptions/tasks/username_and_password.yml for sat63-rhel7

TASK [redhat-subscriptions : Register and Subscribe RHEL Instance to RHN.] ****************************************************************************************************************************************
changed: [sat63-rhel7]

TASK [redhat-subscriptions : include_tasks] ***********************************************************************************************************************************************************************

TASK [manage-repositories : include_tasks] ************************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/manage-repositories/tasks/disable_repositories.yml for sat63-rhel7

TASK [manage-repositories : Install Yum Utils] ********************************************************************************************************************************************************************
changed: [sat63-rhel7]

TASK [manage-repositories : Disable all RHSM repositories] ********************************************************************************************************************************************************
changed: [sat63-rhel7]

TASK [manage-repositories : Disable all Beaker Repos] *************************************************************************************************************************************************************
changed: [sat63-rhel7]

TASK [manage-repositories : include_tasks] ************************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/manage-repositories/tasks/enable_repositories.yml for sat63-rhel7

TASK [manage-repositories : Enable all the required repositories for Satellite6 Installation.] ********************************************************************************************************************
changed: [sat63-rhel7] => (item=rhel-7-server-rpms)
changed: [sat63-rhel7] => (item=rhel-7-server-optional-rpms)
changed: [sat63-rhel7] => (item=rhel-7-server-extras-rpms)
changed: [sat63-rhel7] => (item=rhel-7-server-satellite-maintenance-6-rpms)
changed: [sat63-rhel7] => (item=rhel-7-server-ansible-2-rpms)
changed: [sat63-rhel7] => (item=rhel-server-rhscl-7-rpms)

TASK [manage-repositories : Enable Satellite6 repository.] ********************************************************************************************************************************************************
changed: [sat63-rhel7]

TASK [manage-repositories : Add Satellite 6 repository] ***********************************************************************************************************************************************************

TASK [update-os-packages : yum clean all] *************************************************************************************************************************************************************************
 [WARNING]: Consider using the yum module rather than running yum.  If you need to use command because yum is insufficient you can add warn=False to this command task or set command_warnings=False in
ansible.cfg to get rid of this message.

changed: [sat63-rhel7]

TASK [update-os-packages : Update all installed packages before installing Satellite6] ****************************************************************************************************************************
changed: [sat63-rhel7]

TASK [install-satellite6 : Install the latest version of Satellite6] **********************************************************************************************************************************************
changed: [sat63-rhel7]

PLAY RECAP ********************************************************************************************************************************************************************************************************
sat63-rhel7                : ok=20   changed=14   unreachable=0    failed=0   

```

kbidarkar commented:
Install 6.4 from `internal_repo`:

``` 
kbid  (e) ansible3  ~  git_world  ansible-satellite6  ansible-playbook playbooks/install/satellite_64.yml

PLAY [sat64] ******************************************************************************************************************************************************************************************************

TASK [Gathering Facts] ********************************************************************************************************************************************************************************************
ok: [sat64-rhel7]

TASK [partition-disk : include_tasks] *****************************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/partition-disk/tasks/remove_home_partition.yml for sat64-rhel7

TASK [partition-disk : Unmount /home partition] *******************************************************************************************************************************************************************
ok: [sat64-rhel7]

TASK [partition-disk : Remove /home entry from /etc/fstab] ********************************************************************************************************************************************************
ok: [sat64-rhel7]

TASK [partition-disk : Remove /home Logical Volume] ***************************************************************************************************************************************************************

TASK [partition-disk : include_tasks] *****************************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/partition-disk/tasks/extend_root_partition.yml for sat64-rhel7

TASK [partition-disk : Extend root partition LV to all available VG space] ****************************************************************************************************************************************
ok: [sat64-rhel7] => (item={'key': 'root', 'value': {'size_g': '451.91', 'vg': 'rhel_ibm-x3550m3-07'}})

TASK [partition-disk : Resize the root filesystem] ****************************************************************************************************************************************************************
ok: [sat64-rhel7] => (item={'key': 'root', 'value': {'size_g': '451.91', 'vg': 'rhel_ibm-x3550m3-07'}})

TASK [redhat-subscriptions : include_tasks] ***********************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/redhat-subscriptions/tasks/username_and_password.yml for sat64-rhel7

TASK [redhat-subscriptions : Register and Subscribe RHEL Instance to RHN.] ****************************************************************************************************************************************
changed: [sat64-rhel7]

TASK [redhat-subscriptions : include_tasks] ***********************************************************************************************************************************************************************

TASK [manage-repositories : include_tasks] ************************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/manage-repositories/tasks/disable_repositories.yml for sat64-rhel7

TASK [manage-repositories : Install Yum Utils] ********************************************************************************************************************************************************************
ok: [sat64-rhel7]

TASK [manage-repositories : Disable all RHSM repositories] ********************************************************************************************************************************************************
changed: [sat64-rhel7]

TASK [manage-repositories : Disable all Beaker Repos] *************************************************************************************************************************************************************
changed: [sat64-rhel7]

TASK [manage-repositories : include_tasks] ************************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/manage-repositories/tasks/enable_repositories.yml for sat64-rhel7

TASK [manage-repositories : Enable all the required repositories for Satellite6 Installation via CDN.] ************************************************************************************************************
changed: [sat64-rhel7] => (item=rhel-7-server-rpms)
changed: [sat64-rhel7] => (item=rhel-7-server-optional-rpms)
changed: [sat64-rhel7] => (item=rhel-7-server-extras-rpms)
changed: [sat64-rhel7] => (item=rhel-7-server-ansible-2-rpms)
changed: [sat64-rhel7] => (item=rhel-server-rhscl-7-rpms)

TASK [manage-repositories : Enable Satellite-Maintenance repository via CDN.] *************************************************************************************************************************************

TASK [manage-repositories : Enable Satellite6 repository via CDN.] ************************************************************************************************************************************************

TASK [manage-repositories : Add Internal Satellite-Maintenance repository.] ***************************************************************************************************************************************
changed: [sat64-rhel7]

TASK [manage-repositories : Add Internal Satellite 6 repository.] *************************************************************************************************************************************************
ok: [sat64-rhel7]

TASK [update-os-packages : yum clean all] *************************************************************************************************************************************************************************
 [WARNING]: Consider using the yum module rather than running yum.  If you need to use command because yum is insufficient you can add warn=False to this command task or set command_warnings=False in
ansible.cfg to get rid of this message.

changed: [sat64-rhel7]

TASK [update-os-packages : Update all installed packages before installing Satellite6] ****************************************************************************************************************************
changed: [sat64-rhel7]

TASK [install-satellite6 : Install the latest version of Satellite6] **********************************************************************************************************************************************
changed: [sat64-rhel7]

PLAY RECAP ********************************************************************************************************************************************************************************************************
sat64-rhel7                : ok=20   changed=8    unreachable=0    failed=0  

```

ehelms commented on roles/manage-repositories/tasks/disable_repositories.yml:
```diff
@@ -0,0 +1,19 @@
+---
+# tasks file for manage-repositories
+- name: "Install Yum Utils"
```
Should this just be `yum-utils` given thats the package name?

kbidarkar replied:
Will update it to `Install yum-utils`.

ehelms commented on roles/manage-repositories/tasks/disable_repositories.yml:
```diff
@@ -0,0 +1,19 @@
+# tasks file for manage-repositories
+- name: "Install Yum Utils"
+  when:
+    - ansible_distribution == "RedHat"
```
Given `package` is agnostic of distribution, I think you can drop this

ehelms commented on roles/manage-repositories/tasks/main.yml:
```diff
@@ -0,0 +1,8 @@
+---
```
Just a general comment on role naming. Given a role when applied is an action itself, typically the naming scheme is to use a noun to represent the thing being managed. For example, instead of `manage-repositories` using `satellite6_repositories`.

kbidarkar replied:
ok, will update this, I admit me not good with naming stuff :). I like the `noun to represent` guideline.

kbidarkar replied:
Looking at the role ` manage-repositories` I realized why I gave it the name, it was coz, it had other repos to enable and disable too. But feel better we make it `satellite6-repositories`.

ehelms commented on roles/redhat-subscriptions/tasks/activation_key.yml:
```diff
@@ -0,0 +1,14 @@
+    - ansible_distribution == "RedHat"
+    - ansible_distribution_major_version == "7"
+  yum:
+    name: "{{ repository_url }}/pub/katello-ca-consumer-latest.noarch.rpm"
```
Should this variable be name spaced for the role? `redhat_subscriptions_repository_url` ? Or maybe `redhat_subscriptions_satellite_url` ?

kbidarkar replied:
repository_url is being used in multiple roles, hence using just repository_url.
I believe for `satellite_distribution` and `satellite_version` too as it would be used/required in multiple roles decided not to do that. If any other suggestion around this, let me know.

kbidarkar commented:
Install Sat6.3 from `internal_ak`:
```
 kbid  (e) ansible3  ~  git_world  ansible-satellite6  ansible-playbook playbooks/install/satellite_63.yml 

PLAY [sat63] ******************************************************************************************************************************************************************************************************

TASK [Gathering Facts] ********************************************************************************************************************************************************************************************
ok: [sat63-rhel7]

TASK [partition-disk : include_tasks] *****************************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/partition-disk/tasks/remove_home_partition.yml for sat63-rhel7

TASK [partition-disk : Unmount /home partition] *******************************************************************************************************************************************************************
changed: [sat63-rhel7]

TASK [partition-disk : Remove /home entry from /etc/fstab] ********************************************************************************************************************************************************
changed: [sat63-rhel7]

TASK [partition-disk : Remove /home Logical Volume] ***************************************************************************************************************************************************************
changed: [sat63-rhel7] => (item={'key': 'home', 'value': {'size_g': '1780.38', 'vg': 'rhel_intel-wildcatpass-01'}})

TASK [partition-disk : include_tasks] *****************************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/partition-disk/tasks/extend_root_partition.yml for sat63-rhel7

TASK [partition-disk : Extend root partition LV to all available VG space] ****************************************************************************************************************************************
changed: [sat63-rhel7] => (item={'key': 'root', 'value': {'size_g': '50.00', 'vg': 'rhel_intel-wildcatpass-01'}})

TASK [partition-disk : Resize the root filesystem] ****************************************************************************************************************************************************************
changed: [sat63-rhel7] => (item={'key': 'root', 'value': {'size_g': '50.00', 'vg': 'rhel_intel-wildcatpass-01'}})

TASK [redhat-subscriptions : include_tasks] ***********************************************************************************************************************************************************************

TASK [redhat-subscriptions : include_tasks] ***********************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/redhat-subscriptions/tasks/activation_key.yml for sat63-rhel7

TASK [redhat-subscriptions : Install katello-ca-consumer-latest] **************************************************************************************************************************************************
changed: [sat63-rhel7]

TASK [redhat-subscriptions : Register with activationkey and consume subscriptions from Dogfood Server.] **********************************************************************************************************
changed: [sat63-rhel7]

TASK [manage-repositories : include_tasks] ************************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/manage-repositories/tasks/disable_repositories.yml for sat63-rhel7

TASK [manage-repositories : Install Yum Utils] ********************************************************************************************************************************************************************
changed: [sat63-rhel7]

TASK [manage-repositories : Disable all RHSM repositories] ********************************************************************************************************************************************************

TASK [manage-repositories : Disable all Beaker Repos] *************************************************************************************************************************************************************
changed: [sat63-rhel7]

TASK [manage-repositories : include_tasks] ************************************************************************************************************************************************************************

TASK [update-os-packages : yum clean all] *************************************************************************************************************************************************************************
 [WARNING]: Consider using the yum module rather than running yum.  If you need to use command because yum is insufficient you can add warn=False to this command task or set command_warnings=False in
ansible.cfg to get rid of this message.

changed: [sat63-rhel7]

TASK [update-os-packages : Update all installed packages before installing Satellite6] ****************************************************************************************************************************
changed: [sat63-rhel7]

TASK [install-satellite6 : Install the latest version of Satellite6] **********************************************************************************************************************************************
changed: [sat63-rhel7]

PLAY RECAP ********************************************************************************************************************************************************************************************************
sat63-rhel7                : ok=17   changed=12   unreachable=0    failed=0  

```

kbidarkar commented:
Install Sat6.3 from `internal_repo`:

```
 kbidarka  (e) ansible3  ~  git_world  ansible-satellite6  ansible-playbook playbooks/install/satellite_63.yml 

PLAY [sat63] ******************************************************************************************************************************************************************************************************

TASK [Gathering Facts] ********************************************************************************************************************************************************************************************
ok: [sat63-rhel7]

TASK [partition-disk : include_tasks] *****************************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/partition-disk/tasks/remove_home_partition.yml for sat63-rhel7

TASK [partition-disk : Unmount /home partition] *******************************************************************************************************************************************************************
changed: [sat63-rhel7]

TASK [partition-disk : Remove /home entry from /etc/fstab] ********************************************************************************************************************************************************
changed: [sat63-rhel7]

TASK [partition-disk : Remove /home Logical Volume] ***************************************************************************************************************************************************************
changed: [sat63-rhel7] => (item={'key': 'home', 'value': {'size_g': '406.88', 'vg': 'rhel_cloud-qe-22'}})

TASK [partition-disk : include_tasks] *****************************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/partition-disk/tasks/extend_root_partition.yml for sat63-rhel7

TASK [partition-disk : Extend root partition LV to all available VG space] ****************************************************************************************************************************************
changed: [sat63-rhel7] => (item={'key': 'root', 'value': {'size_g': '50.00', 'vg': 'rhel_cloud-qe-22'}})

TASK [partition-disk : Resize the root filesystem] ****************************************************************************************************************************************************************
changed: [sat63-rhel7] => (item={'key': 'root', 'value': {'size_g': '50.00', 'vg': 'rhel_cloud-qe-22'}})

TASK [redhat_subscriptions : include_tasks] ***********************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/redhat_subscriptions/tasks/username_and_password.yml for sat63-rhel7

TASK [redhat_subscriptions : Register and Subscribe RHEL Instance to RHN.] ****************************************************************************************************************************************
changed: [sat63-rhel7]

TASK [redhat_subscriptions : include_tasks] ***********************************************************************************************************************************************************************

TASK [satellite6_repositories : include_tasks] ********************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/satellite6_repositories/tasks/disable_repositories.yml for sat63-rhel7

TASK [satellite6_repositories : Install yum-utils] ****************************************************************************************************************************************************************
changed: [sat63-rhel7]

TASK [satellite6_repositories : Disable all RHSM repositories] ****************************************************************************************************************************************************
changed: [sat63-rhel7]

TASK [satellite6_repositories : Disable all Beaker Repos] *********************************************************************************************************************************************************
changed: [sat63-rhel7]

TASK [satellite6_repositories : include_tasks] ********************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/satellite6_repositories/tasks/enable_repositories.yml for sat63-rhel7

TASK [satellite6_repositories : Enable all the required repositories for Satellite6 Installation via CDN.] ********************************************************************************************************
changed: [sat63-rhel7] => (item=rhel-7-server-rpms)
changed: [sat63-rhel7] => (item=rhel-7-server-optional-rpms)
changed: [sat63-rhel7] => (item=rhel-7-server-extras-rpms)
changed: [sat63-rhel7] => (item=rhel-7-server-ansible-2-rpms)
changed: [sat63-rhel7] => (item=rhel-server-rhscl-7-rpms)

TASK [satellite6_repositories : Enable Satellite-Maintenance repository via CDN.] *********************************************************************************************************************************

TASK [satellite6_repositories : Enable Satellite6 repository via CDN.] ********************************************************************************************************************************************

TASK [satellite6_repositories : Add Internal Satellite-Maintenance repository.] ***********************************************************************************************************************************
changed: [sat63-rhel7]

TASK [satellite6_repositories : Add Internal Satellite 6 repository.] *********************************************************************************************************************************************
changed: [sat63-rhel7]

TASK [update_os_packages : yum clean all] *************************************************************************************************************************************************************************
 [WARNING]: Consider using the yum module rather than running yum.  If you need to use command because yum is insufficient you can add warn=False to this command task or set command_warnings=False in
ansible.cfg to get rid of this message.

changed: [sat63-rhel7]

TASK [update_os_packages : Update all installed packages before installing Satellite6] ****************************************************************************************************************************
changed: [sat63-rhel7]

TASK [install_satellite6 : Install the latest version of Satellite6] **********************************************************************************************************************************************
changed: [sat63-rhel7]

PLAY RECAP ********************************************************************************************************************************************************************************************************
sat63-rhel7                : ok=21   changed=15   unreachable=0    failed=0 
```
